fennel.lua

```diff
@@ -401,12 +424,14 @@ end
 
 -- Creates Lua-friendly symbol from a single symbol by mangling it.
 local function symMangle(symb)
-    if luaKeywords[symb] then
+    if luaKeywords[symb] or symb:match('^%d') then
         symb = '_' .. symb
     end
-    return symb:gsub('[^%w_]', function(c)
+    local function escaper(c)
         return ('_%02x'):format(c:byte())
-    end)
+    end
+    symb = symb:gsub('-', '_')
```

Comment:
Does this cause dashes to always be converted to underscores? Copying my concerns from https://github.com/bakpakin/Fennel/issues/53#issuecomment-379371973:

> ...that seems really confusing to me, considering that you can use underscores literally in Fennel already. `(let [a-b 1 a_b 2] a-b)` should return `1` IMO. Additionally, that approach doesn't do anything for CamelCase, which in my (limited) experience is more common than snake_case.